### DISCUSSION TITLE
CoreWCF with OData and DataService<T>

### DISCUSSION THREAD

**Discussion by pkudryavtsev** [CoreWCF/CoreWCF#868] [General]
Hi everyone,

Thanks for this project - we are currently considering migrating our WCF endpoints to .NET6 from .NET Framework, and your project looks promising in helping achieving that. 

The project I am working on is huge monolith with custom data access mechanisms which exposes many WCF REST OData services.
However, many of our services are relying on System.Data.Services namespace and implement DataService<T> which supports OData queries. After doing some research, I could not find much on possible equivalent in .NET (Core) for that namespace or other valuable migration paths.

I was wondering whether you may be aware about migration tips that we can utilize to transition our WCF REST OData services to .NET6 be it with CoreWCF or without, or other ideas such as completely abandoning WCF in favor of WebApi controllers for example. Would be great to hear your thoughts!

Regards,
Pavlo

**Comment by eocrawford** [comment]
Opened https://github.com/CoreWCF/CoreWCF/issues/1099 to track
